File: POO/php/Calculadora.php
<?php
class Calculadora {
    // declarar atributos
    private $valorBase;
    private $porcentajeImpuesto;
    private $valorTotal;
    private $cantidadImpuesto;
    // Inicializar atributos
    public function __construct() {
        $this->valorBase = null;
        $this->porcentajeImpuesto = null;
        $this->valorTotal = null;
        $this->cantidadImpuesto = null;
    }
    // Get
    public function darValorBase() {
        return $this->valorBase;
    }
    // Set
    public function recibirValorBase($pValorBase) {
        $this->valorBase = $pValorBase;
    }
    // Get
    public function darPorcentajeImpuesto() {
        return $this->porcentajeImpuesto;
    }
    // Set
    public function recibirPorcentajeImpuesto($pPorcentajeImpuesto) {
        $this->porcentajeImpuesto = $pPorcentajeImpuesto;
    }
    // Get
    public function darValorTotal() {
        return $this->valorTotal;
    }
    // Set
    public function recibirValorTotal($pValorTotal) {
        $this->valorTotal = $pValorTotal;
    }
    // Get
    public function darCantidadImpuesto() {
        return $this->cantidadImpuesto;
    }
    // Set
    public function recibirCantidadImpuesto($pCantidadImpuesto) {
        $this->cantidadImpuesto = $pCantidadImpuesto;
    }
    // Mundo del problema
    public function calcularValorBase() {
        $this->valorBase = $this->valorTotal / (1 + $this->porcentajeImpuesto / 100);
        $this->calculalarCantidadImpuesto();
    }
    public function calcularValorTotal() {
        $this->valorTotal = $this->valorBase * (1 + $this->porcentajeImpuesto / 100);
        $this->calculalarCantidadImpuesto();
    }
    public function calculalarCantidadImpuesto() {
        $this->cantidadImpuesto = $this->valorTotal - $this->valorBase;
    }
}
?>

File: POO/py/Calculadora.py
class Calculadora:
    def __init__(self):
        self.__valorBase = 0
        self.__porcentajeImpuesto = 0
        self.__valorTotal = 0
        self.__cantidadImpuesto = 0

    def darValorBase(self):
        return self.__valorBase

    def recibirValorBase(self, pValorBase):
        self.__valorBase = pValorBase

    def darPorcentajeImpuesto(self):
        return self.__porcentajeImpuesto

    def recibirPorcentajeImpuesto(self, pPorcentajeImpuesto):
        self.__porcentajeImpuesto = pPorcentajeImpuesto

    def darValorTotal(self):
        return self.__valorTotal

    def recibirValorTotal(self, pValorTotal):
        self.__valorTotal = pValorTotal

    def darCantidadImpuesto(self):
        return self.__cantidadImpuesto

    def recibirCantidadImpuesto(self, pCantidadImpuesto):
        self.__cantidadImpuesto = pCantidadImpuesto

    # Mundo del problema

    def calcularValorBase(self):
        self.__valorBase = self.__valorBase = self.__valorTotal / \
            (1 + self.__porcentajeImpuesto / 100)
        self.calcularCantidadImpuesto()

    def calcularValorTotal(self):
        self.__valorTotal = self.__valorTotal = self.__valorBase * \
            (1 + self.__porcentajeImpuesto / 100)
        self.calcularCantidadImpuesto()

    def calcularCantidadImpuesto(self):
        self.__cantidadImpuesto = self.__valorTotal - self.__valorBase
